python/cudaq/kernel/ast_bridge.py

```diff
@@ -1469,6 +1469,24 @@ def bodyBuilder(iterVal):
                 opCtor([], [], [], [qubitA, qubitB])
                 return
 
+            if node.func.id == 'reset':
+                target = self.popValue()
+                # single qubit
+                if not quake.VeqType.isinstance(target.type):
+                    quake.ResetOp([], target)
+                    return
+                # target is a VeqType
+                def bodyBuilder(iterVal):
+                    q = quake.ExtractRefOp(self.getRefType(),
+                                           target,
+                                           -1,
```

Comment:
This magic value looks a bit odd. I believe it corresponds to `quake::ExtractOp::kDynamicIndex`, which isn't `-1` but has the same bits as `-1`.
```suggestion
                                           -1, # kDynamicIndex
```

On the C++ side, we have both a constructor for using a constant index and another for using an SSA-value as an index. The internal encoding isn't exposed in this way. It might make sense to do something similar in python.